src/resolve/resolved-link-bus.c

```diff
@@ -465,6 +465,10 @@ int bus_link_method_set_default_route(sd_bus_message *message, void *userdata, s
         if (l->default_route != b) {
                 l->default_route = b;
 
+                DnsServer *server = l->manager->current_dns_server;
+                if (server && server->type == DNS_SERVER_FALLBACK)
+                        manager_set_dns_server(l->manager, NULL);
```

Comment:
The same thing needs to be done also in `link_update_default_route()`.

Maybe, better to introduce a tiny helper function `link_set_default_route(Link *l, bool b)` with comments why we need to unset the current DNS server if it is fallback one.